### DISCUSSION TITLE
Segmentation fault and STL files to MD

### DISCUSSION THREAD

**Discussion by MishterBluesky** [weria-pezeshkian/FreeDTS#5] [General]
I am trying to create a membrane system of a custom size, between two planes, that have a channel between them, however so far i have had segmentation faults in both TS2CG and FreeDTS when attempting to simulate or add lipids, at the PLM step and at the first attempts with FreeDTS. 

However simple sphere files seem to work.

What might be causing these issues?

**Comment by weria-pezeshkian** [comment]
Could you send the files and also write what you want to do step by step? Because it is not clear what you mean. Also what kind of error messages do you get?

**Reply by MishterBluesky** [reply]
Thank you a lot for replying it's really appreciated. Thank you for a great piece of software.

Ive created an STL file of what Id eventually like to achieve, which is to see how we might make a more stable tunnel between membranes, and have used blender to make various tunnel sizes dependent on EM data measurements, Then Ive taken these STL files, and turned them into a .tsi file, which ive attached as a .txt. (for ease of sharing)

[cleantunnel.txt](https://github.com/weria-pezeshkian/FreeDTS/files/12072862/cleantunnel.txt) - a curved cylinder pressed against two membranes, I have also tried the same system without the adjacent membranes (only cylinder)
[closed.txt](https://github.com/weria-pezeshkian/FreeDTS/files/12072863/closed.txt) - closed system much larger

When I use this .tsi file , or files like it, which Ive created closed systems of later on in STL [closed.txt](https://github.com/weria-pezeshkian/FreeDTS/files/12072836/closed.txt), I get segmentation errors. - Ive also tried smaller system sizes.
I would like to be able to use FreeDTS, or TS2CG modules to be able to first simulate this, then attempt coarse grain of these shapes later on, especially if conversions from the .STL files go well. - Ive tried various remeshing tools to improve the meshes, this may also be an issue.

I would like to:

1. Take .tsi files made from custom .STL meshes and either 
a) simulate bilayers along the triangle faces these .STL files described by .tsi files directly in FreeDTS, with or without protein inclusions b) Use TS2CG pointilism to place a lipid bilayer along the .STL faces, for later simulation (4.3nm)
2. Evaluate stability with various conditions 

If this is not possible, creating a similar system using the FreeDTS system to create a similar architecture without blender would also be interesting, for example placing a cylinder and two flat membranes adjacent either side (different .tsi files) and creating the same tunnels this way somehow.

I very much appreciate the help! Apologies for the longer response. 

<img width="612" alt="image" src="https://github.com/weria-pezeshkian/FreeDTS/assets/19994410/b025b2c1-2e60-47b3-82cc-7bdb9fe05d23">

**Comment by weria-pezeshkian** [comment]
You can always check a tsi file by using the CNV script in FreeDTS as
CNV -in cleantunnel.tsi -o conf.vtu

and then use paraview to look at conf.vtu file (see below for the figures).
while your  points seems to be in ok positions, your triangles are missed-up. also, the box is way too big. 

You can also use vmd and download https://github.com/weria-pezeshkian/TS2CG1.2/blob/master/ts2vmd.tcl 
From vmd console
write:

source ts2vmd.tcl
load_ts -in cleantunnel.tsi
show_surf -mesh edge -color gray

You can see the edges and triangles are not right.

I would suggest to re-mash you your points, and use this version of TS2CG 
https://github.com/weria-pezeshkian/TS2CG1.2
This one allows you to use meshes with free edges.    


<img width="634" alt="Screenshot 2023-07-17 at 21 36 50" src="https://github.com/weria-pezesh
<img width="639" alt="Screenshot 2023-07-17 at 21 36 43" src="https://github.com/weria-pezeshkian/FreeDTS/assets/47776510/9d71af2c-56d5-4a32-adc5-e382d0b0900a">

<img width="634" alt="Screenshot 2023-07-17 at 21 36 50" src="https://github.com/weria-pezeshkian/FreeDTS/assets/47776510/bf12829c-0c11-4dc7-bcac-8122111b5fe8">

**Reply by MishterBluesky** [reply]
Oof, thank you for your reply, I then later tried your advise, and fixed my .tsi filemaker which i had been modifying a bit too much. 

In the end getting a nicer result.

I also with your advise made a new way to form the .stl files required.

<img width="482" alt="image" src="https://github.com/weria-pezeshkian/FreeDTS/assets/19994410/ac14417a-4d69-4b9f-a028-7bb9a2047b01">

You have solved the issue it seems, no more segmentation faults.

Very best wishes

Chris

**Reply by MishterBluesky** [reply]
ignore the headless channel though..

**Reply by weria-pezeshkian** [reply]
I think still there are some issues with the tsi file as the output look a bit rough. Try MD and see if you can equilibrate it.
